lib3mf PR integration

baptistapedro opened the pull request.

Commit:
Create Dockerfile

Commit:
Create Mayhemfile

Commit:
Update Mayhemfile

Commit:
Create mayhem.yml

Commit:
Update Mayhemfile

Commit:
Delete build.yml

Commit:
Update Dockerfile

Commit:
Update Dockerfile

Commit:
Create fuzz.cpp

Commit:
Update Dockerfile

xansec commented on Dockerfile:
```diff
@@ -0,0 +1,24 @@
+FROM fuzzers/afl:2.52
```
Please don't use git clone here. In fact I don't think you need to do wget either (since the repo is already in the Docker image). I'm not sure I understand the purpose of LD_LIBRARY_PATH being set. Finally, please include a package stage.

baptistapedro replied:
these are a little bit old harnesses, need indeed refactoring. LD_LIBRARY_PATH is required because when run in Mayhem it fails because can not find the lib3mf shared object. Mayhem also recommends to set it.

xansec commented on fuzzers/fuzz.cpp:
```diff
@@ -0,0 +1,241 @@
+
```
Did you copy these functions from the source code? If so, they are static copies and will not track changes if the upstream functionality is changes. Please link in or include the functions rather than copying them. Also, please rename this from `fuzz.cpp` to something that describes what's being fuzzed? Also, I don't think using std::cout is the best way to read a file in. Maybe you could use libFuzzer here? I also recommend not having any std:cout for errors to improve fuzzing speed.

baptistapedro replied:
Yes it is an example code reading from stdin,  most of the functions are printing stuff so I just refactored the harness to be fuzz_3mfReader.cpp.
std::cout calls deleted as well.

xansec reviewed (changes requested):
Please don't delete upstream files. Please make the requested changes.

Commit:
Update Dockerfile

Commit:
Update and rename fuzz.cpp to fuzz_3mfReader.cpp

Commit:
Update Dockerfile

baptistapedro commented:
About deleting upstream files, the only files I delete in the upstream are the Actions as I had experiences of having them failing, plus receiving emails about it and were not related to Mayhem.

Commit:
Update fuzz_3mfReader.cpp

Commit:
Update Dockerfile

Commit:
Create build.yml

Adding upstream Action file previously deleted. No effects to Mayhem functionality.
